ton_sdk/src/contract.rs

```diff
@@ -51,6 +52,110 @@ use ton_block::TransactionProcessingStatus;
 use serde_json::Value;
 #[cfg(feature = "node_interaction")]
 use futures::FutureExt;
+use ton_vm::stack::{StackItem, Stack};
+use ton_vm::stack::integer::IntegerData;
+use std::sync::Arc;
+
+// JSON extension to StackItem
+//
+struct StackItemJSON;
+
+impl StackItemJSON {
+    fn invalid_json() -> SdkError {
+        SdkError::InvalidData { msg: "Invalid JSON value for stack item".to_owned() }
+    }
+
+    fn json_array_from_items(items: Iter<StackItem>) -> Result<Value> {
+        let mut values = Vec::<Value>::new();
+        for item in items {
+            values.push(StackItemJSON::json_value_from_item(item)?)
+        }
+        Ok(Value::Array(values))
+    }
+
+    fn items_from_json_array(values: Iter<Value>) -> Result<Vec<StackItem>> {
+        let mut items = Vec::<StackItem>::new();
+        for value in values {
+            items.push(Self::item_from_json_value(value)?)
+        }
+        Ok(items)
+    }
+
+    fn json_value_from_item(item: &StackItem) -> Result<Value> {
+        Ok(match item {
+            StackItem::None =>
+                Value::Null,
+            StackItem::Integer(i) => {
+                let mut hex = i.to_str_radix(16);
+                if hex.ne("NaN") {
+                    hex.insert_str(if hex.starts_with("-") { 1 } else { 0 }, "0x")
+                }
+                Value::String(hex)
+            }
+            StackItem::Tuple(items) =>
+                Self::json_array_from_items(items.iter())?,
+            StackItem::Builder(_) =>
+                json!({"builder": Value::Null}),
+            StackItem::Slice(_) =>
+                json!({"slice": Value::Null}),
+            StackItem::Cell(_) =>
+                json!({"cell": Value::Null}),
+            StackItem::Continuation(_) =>
+                json!({"continuation": Value::Null}),
+        })
+    }
+
+    fn parse_integer_data(s: &String) -> Result<IntegerData> {
+        Ok(if s.eq("NaN") {
+            IntegerData::nan()
+        } else {
+            let hex_prefix_pos =
+                if s.starts_with("0x") || s.starts_with("0X") {
+                    Some(0)
+                } else if s.starts_with("-0x") || s.starts_with("-0X") {
+                    Some(1)
+                } else {
+                    None
+                };
+            if let Some(pos) = hex_prefix_pos {
+                let mut ms = s.clone();
+                ms.remove(pos);
+                ms.remove(pos);
+                IntegerData::from_str(ms.as_str())?
```

Comment:
`IntegerData::from_str(&s[pos..])?` and remove `ms` variable

Reply:
It's not the same.
"-0xabcd" must convert to "-abcd"